docs: edits/xrefs for keptnapp migration (#2944)
keptn/lifecycle-toolkit: docs/docs/migrate/keptnapp/index.md

# Migrating KeptnApp to KeptnAppContext

The
[KeptnAppContext](../../reference/crd-reference/appcontext.md)

> resource is new in the `v1beta1` API included in Keptn v0.11.0.

Reviewer: I think the last KLT is [klt: v0.9.0](https://github.com/keptn/lifecycle-toolkit/releases/tag/klt-v0.9.0) so the next would be 0.10.0 ... should we try to keep this easy to mantain? So instead of mentioning v1beta1, say something as is introduced from v0.10.0?

Author: Oh my!  I was sure we did a 0.10.0 release.  I just spent a significant amount of time sniffing around the web site trying to figure out what the latest release is...  Finally found something that shows metrics-operator 0.9.0 but the most recent lifecycle-operator is 0.8.3!  Maybe the group needs a discussion about how we reference releases and how a user knows which friggin' release they are running?  For now, maybe the docs should just reference the API version?

Author: I removed the release number for now.

Existing
[KeptnApp](../../reference/crd-reference/app.md)
resources that were generated manually must be migrated
because of this new feature..
This page gives instructions for doing that.

> **Note**
> Manual migration is only required for:
>
> * Manually created `KeptnApp` resources
> * Automatically created `KeptnApp` resources with
>   manually edited pre/post-deployment tasks or evaluations**
>
> `KeptnApp` resources created using the
> [automatic app-discovery](../../guides/auto-app-discovery.md)
> feature without any manual changes in the pre/post-deployment
> tasks/evaluations section do not require any action.

## Migration steps

The steps to migrate an existing `KeptnApp` resource

> to work with the `KeptnAppContext` resource introduced in version 0.11.0,

Reviewer: ```suggestion
to work with the `KeptnAppContext` resource introduced in version 0.10.0,
```

you need to execute the following steps:

1. Create a `KeptnAppContext` custom resource
   that has the same name as your `KeptnApp`.
2. Move the lists of pre/post-deployment tasks and evaluations
   from `KeptnApp` to `KeptnAppContext`.
   In other words, delete them from `KeptnApp.spec`
   and add them under the `KeptnAppContext.spec` field.

> 3. If necessary, add the `app.kubernetes.io/managed-by: keptn` annotation

Reviewer: ```suggestion
3. If not present already, add the `app.kubernetes.io/managed-by: keptn` annotation
```

Reviewer: or something like that?

Author: done

# Migrating KeptnApp to KeptnAppContext

The
[KeptnAppContext](../../reference/crd-reference/appcontext.md)
resource is new in the `v1beta1` API.
Existing
[KeptnApp](../../reference/crd-reference/app.md)
resources that were generated manually must be migrated
because of this new feature..
This page gives instructions for doing that.

> **Note**
> Manual migration is only required for:
>
> * Manually created `KeptnApp` resources
> * Automatically created `KeptnApp` resources with
>   manually edited pre/post-deployment tasks or evaluations**
>
> `KeptnApp` resources created using the
> [automatic app-discovery](../../guides/auto-app-discovery.md)
> feature without any manual changes in the pre/post-deployment
> tasks/evaluations section do not require any action.

## Migration steps

> The steps to migrate an existing `KeptnApp` resource

Reviewer: ```suggestion
You need the following steps to migrate an existing `KeptnApp` resource
```

to work with the `KeptnAppContext` resource

> introduced in the `v1beta1` API version.

Reviewer: ```suggestion
introduced in the `v1beta1` API version:
```

> you need to execute the following steps:

Reviewer 2: this is not a full sentence

> you need to execute the following steps:

Reviewer: ```suggestion
```

{% include "./assets/keptnapp.yaml" %}
```

Applying the migration steps from the previous subsection,

> we get the following result.

Reviewer: ```suggestion

```

Author: fixed

> You see the following changes:

Reviewer: ```suggestion
you see the following changes:
```

* The `app.kubernetes.io/managed-by` annotation
  has been added to the `labels` section of the revised `KeptnApp` resource.

     If your original `KeptnApp` resource was auto-generated,
     it already had this annotation.

* The list of tasks and evaluations
has been moved from the `KeptnApp` resource
to the `KeptnAppContext` resource

```yaml
{% include "./assets/keptnapp-migrated.yaml" %}
```

These modified resources can be now applied to your cluster.

## What's next?

Making these modifications allows your existing `KeptnApp` functionality

> run as it did before.

Reviewer: this sentence sounds strange 🤔 maybe something stating these changes do not alter Keptn's behavior?

Author: I reworded it.

However, new capabilities can be added:

* Add context metadata to your traces.

>   This allows you to include information like `stage` in your workload traces
>   and information such as commit ID or user name

Reviewer 3: Maybe we can rephrase this a bit to make it clear that the attributes you can add are not limited to the examples we provide here, so something like:

```suggestion
  This allows you to include information like the stage
  the application is deployed into, a commit ID,
  or any other information relevant to the
  deployment to the deployment traces of
  the application and its workloads.
```

>   in your `KeptnApp` traces.

Reviewer: And all associated Keptn Workloads ?

Author: Flo's suggestion fixed this, too.

{% include "./assets/keptnapp.yaml" %}
```

After applying the migration steps from the previous subsection,
you see the following changes:

* The `app.kubernetes.io/managed-by` annotation

>   has been added to the `labels` section of the revised `KeptnApp` resource.

Reviewer 2: ```suggestion
  has been added to the `metadata.labels` section of the revised `KeptnApp` resource.
```

     If your original `KeptnApp` resource was auto-generated,
     it already had this annotation.

* The list of tasks and evaluations
has been moved from the `KeptnApp` resource

> to the `KeptnAppContext` resource

Reviewer 2: ```suggestion
to the `KeptnAppContext` resource.
```
